src/braket/experimental/autoqasm/doc/decorators.md

```diff
@@ -8,26 +8,26 @@ There are a handful of decorators available through AutoQASM. Each one attaches
 
 This decorator marks the entry point to a quantum program.
 
-You can include gates, pulse control, classical control and subroutine calls. When you call the function wrapped by `@aq.main`, you will get a `Program` object. The `Program` object can be executed on [devices available through Amazon Braket](https://docs.aws.amazon.com/braket/latest/developerguide/braket-devices.html), including local simulators. The code snippet below creates a quantum program with `@aq.main` and runs it on the `Device` instantiated as `device`.
+You can include gates, pulse control, classical control and subroutine calls. The function wrapped by `@aq.main` is converted into a `Program` object. The `Program` object can be executed on [devices available through Amazon Braket](https://docs.aws.amazon.com/braket/latest/developerguide/braket-devices.html), including local simulators. The code snippet below creates a quantum program with `@aq.main` and runs it on the `Device` instantiated as `device`.
```

Comment:
Reading this really makes me feel that the wrapped function (the Program object) should be callable and should "run the wrapped function" when called. I think it would be good to use `my_program(**inputs)` to run on a device.

If a device isn't set in the decorator, we have multiple options with different tradeoffs for setting later:
- defaulting to local sim (do people want this? maybe? would require setting other devices in the decorator)
- using device as a reserved keyword arg for calling: `my_program.run(**inputs, device=local_sim)` (name conflicts)
- setter method: `my_program.on(Devices.Amazon.SV1)(**inputs)` (clunky?)
- set and run method: `my_program.run_on(Devices.Amazon.SV1, **inputs)` (not bad, but not utilizing the call method)

Not related to this PR, but emotionally related to this portion of docs

Reply:
These are good points - added this to the [issue tracking this](https://github.com/orgs/amazon-braket/projects/2/views/1?pane=issue&itemId=45874986).